apps/web-giddh/src/app/services/sales.service.ts

```diff
@@ -113,4 +115,47 @@ export class SalesService {
             return data;
         }), catchError((e) => this.errorHandler.HandleCatch(e)));
     }
+
+    /**
+     * API call to get all advance receipt vouchers using invoice date
+     *
+     * @param {AdvanceReceiptRequest} model Request model
+     * @returns {Observable<BaseResponse<any, AdvanceReceiptRequest>>} API response
+     * @memberof SalesService
+     */
+    public getAllAdvanceReceiptVoucher(model: AdvanceReceiptRequest): Observable<BaseResponse<any, AdvanceReceiptRequest>> {
+        this.user = this._generalService.user;
+        this.companyUniqueName = this._generalService.companyUniqueName;
+        return this._http.get(this.config.apiUrl + ADVANCE_RECEIPTS_API.GET_ALL_ADVANCE_RECEIPTS
+            .replace(':companyUniqueName', this.companyUniqueName)
+            .replace(':accountUniqueName', model.accountUniqueName).replace(':invoiceDate', model.invoiceDate))
+            .pipe(
+                map((res) => {
+                    let data: BaseResponse<any, AdvanceReceiptRequest> = res;
+                    data.request = model;
+                    return data;
+                }),
+                catchError((e) => this.errorHandler.HandleCatch<any, AdvanceReceiptRequest>(e, model)));
+    }
+
+    /**
+     * API call to adjust an invoice with advance receipts
+     *
+     * @param {AdvanceReceiptAdjustment} model Adjust advance receipts request model
+     * @param {string} invoiceUniqueName Invoice unique name which need to adjust
+     * @returns {Observable<BaseResponse<any, any>>}
+     * @memberof SalesService
+     */
+    public adjustAnInvoiceWithAdvanceReceipts(model: AdvanceReceiptAdjustment, invoiceUniqueName: string): Observable<BaseResponse<any, any>> {
+        this.user = this._generalService.user;
+        this.companyUniqueName = this._generalService.companyUniqueName;
```

Comment:
Don't use companyUniqueName from general service. It will come wrong if we will switch company.

Reply:
Have change in header